libsplinter/src/admin/rest_api/actix/ws_register_type.rs

```diff
@@ -31,12 +33,134 @@ use crate::rest_api::{
 pub fn make_application_handler_registration_route<A: AdminCommands + Clone + 'static>(
     admin_commands: A,
 ) -> Resource {
-    Resource::build("/ws/admin/register/{type}")
-        .add_request_guard(ProtocolVersionRangeGuard::new(
+    let resource = Resource::build("/ws/admin/register/{type}").add_request_guard(
+        ProtocolVersionRangeGuard::new(
             protocol::ADMIN_APPLICATION_REGISTRATION_PROTOCOL_MIN,
             protocol::ADMIN_PROTOCOL_VERSION,
-        ))
-        .add_method(Method::Get, move |request, payload| {
+        ),
+    );
+
+    #[cfg(feature = "authorization")]
+    {
+        resource.add_method(
+            Method::Get,
+            CIRCUIT_READ_PERMISSION,
```

Comment:
I'm actually curious if people think this should be a write permission

Reply:
It is just a read only stream